### DISCUSSION TITLE
Can't access Admin Console when using reverse proxy

### DISCUSSION THREAD

**Discussion by eucleciojosias** [keycloak/keycloak#11550] [Archive]
My Dockerfile:
```dockerfile
FROM quay.io/keycloak/keycloak:nightly as builder

ENV KC_FEATURES=authorization,account2,account-api,admin-fine-grained-authz,admin2,docker,impersonation,token-exchange,client-policies,declarative-user-profile,dynamic-scopes,preview
ENV KC_DB=postgres

RUN /opt/keycloak/bin/kc.sh build

FROM quay.io/keycloak/keycloak:nightly

COPY --from=builder /opt/keycloak/lib/quarkus/ /opt/keycloak/lib/quarkus/

WORKDIR /opt/keycloak

ARG KEYCLOAK_ADMIN
ARG KEYCLOAK_ADMIN_PASSWORD
ARG KC_DB_URL
ARG KC_DB_USERNAME
ARG KC_DB_PASSWORD
ARG KC_LOG_LEVEL

ENV KEYCLOAK_ADMIN=$KEYCLOAK_ADMIN
ENV KEYCLOAK_ADMIN_PASSWORD=$KEYCLOAK_ADMIN_PASSWORD
ENV KC_DB_URL=$KC_DB_URL
ENV KC_DB_USERNAME=$KC_DB_USERNAME
ENV KC_DB_PASSWORD=$KC_DB_PASSWORD
ENV KC_LOG_LEVEL=$KC_LOG_LEVEL

ENV KC_PROXY=edge
ENV KC_HOSTNAME=keycloak.betrybe.dev

ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start"]
```

I tried so many approaches, but a just can't access the admin console. 

I access `https://my.domain.dev` then I click in **Administration Console** that goes to `https://my.domain.dev/admin/master/console/` with a loading **Loading the admin console**, then show up a js alert with a message `failed to initialize keycloak` the it's redirected to a error page with the URL: 
`https://my.domain.dev/realms/master/protocol/openid-connect/auth?client_id=security-admin-console&redirect_uri=https%3A%2F%2Fmy.domain.dev%2Fadmin%2Fmaster%2Fconsole%2F%23%2F&state={}&response_mode=fragment&response_type=code&scope=openid&nonce={}&code_challenge={}&code_challenge_method=S256`

With the error `invalid_url` in the image below 
<img width="736" alt="Screen Shot 2022-04-20 at 16 18 18" src="https://user-images.githubusercontent.com/46788263/164307406-e5723961-9642-4f36-a5a7-02dc4d432f9b.png">

**Comment by Paul6552** [comment]
Had the same problem: [https://keycloak.discourse.group/t/migration-from-docker-15-to-17-0-1/14878/2](Migration from 15 to 17)

I think you are missing two properties: hostname-strict: false and hostname-strict-https=false

**Comment by chinnayya-nalla** [comment]
@eucleciojosias  could please let me know how you have fixed this problem ? i am facing the same issue

**Comment by patelke1** [comment]
Having a similar problem with no reverse proxy. Just accessing with a different name and when  I click on Admin Console, get the same message. Tried hostname-strict=false and did not work. Anyone have any other suggestions.

**Reply by jrivers96** [reply]
I think 18.0.1 has a fix for this. I'm waiting on the solution as well. 

I think they add a new environment variable. 

 - name: KC_SPI_HOSTNAME_DEFAULT_ADMIN
    value: "keycloak.admin.dev.yourhostname.com"

**Comment by XcrigX** [comment]
Not sure if this is anyone's problem here - but see the last post here: 
https://github.com/keycloak/keycloak/discussions/12466

It seems that they logon dialog for the admin uses the public hostname even when accessing keycloak using a private hostname - or at least it does with the various configuration settings I am using.
